homeassistant/components/binary_sensor/zoneminder.py

```diff
@@ -0,0 +1,66 @@
+"""
+Support for state (idle, alarm) of ZoneMinder monitors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.zoneminder/
+"""
+import logging
+from homeassistant.components.binary_sensor import (BinarySensorDevice)
+import homeassistant.components.zoneminder as zoneminder
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['zoneminder']
+ZM_IDLE = '0'
+ZM_TAPE = '4'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ZoneMinder sensors."""
+    sensors = []
+
+    _LOGGER.debug("Getting monitors")
+    monitors = zoneminder.get_state('api/monitors.json')
+    for i in monitors['monitors']:
+        sensors.append(
+            ZMBinarySensor(int(i['Monitor']['Id']), i['Monitor']['Name'])
+        )
+
+    _LOGGER.debug("Adding devices")
+    add_devices(sensors)
+
+
+class ZMBinarySensor(BinarySensorDevice):
+    """Get the state of each ZoneMinder monitor."""
+
+    def __init__(self, monitor_id, monitor_name):
+        """Initiate monitor sensor."""
+        self._monitor_id = monitor_id
+        self._monitor_name = monitor_name
+        self._sensor_class = 'motion'
+        self._state = False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} Status'.format(self._monitor_name)
+
+    @property
+    def is_on(self):
+        """Return the status of the sensor."""
+        _LOGGER.debug('is on?')
+        return self._state
+
+    def update(self):
+        """Return the current status of the monitor."""
+        _LOGGER.debug("getting status")
+        monitor = zoneminder.get_state(
+            'api/monitors/alarm/id:%i/command:status.json' % self._monitor_id
+        )
+        _LOGGER.debug("got status")
+        if monitor['status'] == ZM_IDLE:
+            self._state = False
+        elif monitor['status'] == ZM_TAPE:
+            self._state = False
+        else:
+            self._state = 'Alarm'
```

Comment:
Hi Matt,

I'll start off with:  I wasn't expecting to be dealing with someone that knew how ZoneMinder worked so well.  Kudos!

I think that the edge case here is with prealarm, as only those people that have configured it on their ZM system are going to have to worry about that false positive (in the binary sensor context).  Prealarm is definitely useful for cases such as yours where you have only a few frames changing such as when the camera cuts over.

I'll admit that I wasn't as clear as I thought I was on how prealarm worked (I was confusing it with the warmup buffer, for some reason) when I created this PR, and in hindsight, it would probably be best to ignore that state.

ALERT is useful, I think, in this case, where we are polling from ZM (instead of pushed to HASS).  If you don't poll at the right interval, you could completely miss all non-0 states.  By setting motion on ALERT and ALARM, that gives HASS some more leeway in how often it has to poll ZM in order to catch an ongoing event.

For my own case, I care about only two things: motion, or not?  Prealarm aside, I could simply say that STATE_ALARM || STATE_ALERT = on, all others = off, or if we really wanted to be picky, ALARM = on, all others = off.